### PR TITLE
Add blocks to moosepy

### PR DESCRIPTION
**Pull request by DatGuyJonathan**
- Adds blocks to lib and use it in default project
- Adds MooseClient to consumption api example
- Adds cliLog to lib. Can be used like this:

```
from moose_lib import cli_log, CliLogData

log_data = CliLogData(action="UserLogin", message="User has logged in successfully.")
cli_log(log_data)
```

### CODE REVIEW

**Review by phiSgr — COMMENTED** [review]
Upper case `List` is deprecated
https://docs.python.org/3/library/typing.html#typing.List

File: packages/py-moose-lib/moose_lib/main.py [hunk 1]
```diff
@@ -10,6 +11,26 @@
...
+#     JSON_ARRAY = "JSON_ARRAY"
+
+# T = TypeVar('T')
+
+# @dataclass
+# class IngestionConfig:
+#     format: Optional[IngestionFormat] = None
+
+# @dataclass
+# class StorageConfig(Generic[T]):
```
Comment (phiSgr) [inline packages/py-moose-lib/moose_lib/main.py:25]: In TS I was trying to be cute and used `keyof T` for `order_by_fields`. Python probably can't do this, so we don't need the generic type param.

**Review by phiSgr — COMMENTED** [review]
LGTM other than the upper case `List`

**Comment by DatGuyJonathan** [comment]
> Upper case `List` is deprecated https://docs.python.org/3/library/typing.html#typing.List

Good call, I've updated it.

**Review by phiSgr — APPROVED** [review]

**Pull request merged**